tests/queries/0_stateless/01529_union_distinct_and_setting_union_default_mode.sql

```diff
@@ -0,0 +1,13 @@
+SELECT * FROM numbers(10) UNION SELECT * FROM numbers(10);
+SELECT * FROM numbers(10) UNION ALL SELECT * FROM numbers(10);
+SELECT * FROM numbers(10) UNION DISTINCT SELECT * FROM numbers(10);
+
+SET union_default_mode='ALL';
```

Comment:
It would be helpful to add something like `select 'ALL'` here, so that there is some visual separation in the reference file, and not only numbers. This makes it easier to understand where is the difference, when the test breaks.

Reply:
Thanks for review! This task has not been finished yet. 

It 's not very easy for me,  I found we cann't simply flatten all subqueries to select lists and build query plan now, because simply flatten subqueries to select lists may change query's semantics.
For example, flatten `SELECT 1 UNION (SELECT 1 UNION ALL SELECT 1) -> SELECT 1 UNION SELECT 1 UNION ALL SELECT 1` changes query's semantics.

I will rewrite test when code finished.

Reply:
Hi, I have finished this task, and rewrite test. Thanks for review! @akuzm